configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyConfiguration.java

```diff
@@ -134,6 +134,11 @@ public boolean wasKeyProvided(@NotNull final String key) {
         return resolveFromCache(property.getKey()).getResolutionInfo().map(PropertyResolutionInfo::getSource);
     }
 
+    public Optional<String> getPropertySource(@NotNull final String key) {
```

Comment:
This doesn't seem necessary if the other comment is addressed, though I think it's fine overall.